sdk/src/main/java/io/dapr/client/DaprErrorResponseParser.java

```diff
@@ -0,0 +1,8 @@
+package io.dapr.client;
+
+import io.dapr.exceptions.DaprException;
+import java.io.IOException;
+
+public interface DaprErrorResponseParser {
+    DaprException parse(int statusCode, byte[] response) throws IOException;
```

Comment:
What is statusCode? These are different for gRPC or HTTP. Maybe add a new param for the protocol, so the implementation can handle both. Another option would be to have one method for gRPC and one for HTTP.

Reply:
My intention with the DaprErrorResponseParser interface was to provide separate implementations for handling different protocols (as demonstrated by DefaultDaprHttpErrorResponseParser and DefaultDaprGrpcErrorResponseParser). I believe this approach maintains a clean separation of concerns between gRPC and HTTP error handling, which is why I opted not to include a protocol parameter in the interface. Ofc if you think there is a strong reason for having a single implementation handling both protocols, then I can change it